fury/tests/test_actors.py

```diff
@@ -945,8 +950,12 @@ def test_advanced_geometry_actor(interactive=False):
         heights = 10
 
         scene.clear()
-        geom_actor = act_func(centers=xyz[:, :3], heights=10, colors=colors[:],
-                              **extra_args)
+        if act_func == actor.disk:
```

Comment:
Testing the orientation argument is not possible without a test on repeat_sources. Also, I would argue against creating a separate test for the disk actor, since it is an advanced geometry actor.
I've added arguments to test the actor-specific parameters, though.

Reply:
The purpose of the separate test is to remove the unnecessary ifs. Another way to have a single test that still looks clean is to do something like this:

```python
def test_advanced_geometry_actor(interactive=False):
    xyz = np.array([[0, 0, 0], [50, 0, 0], [100, 0, 0]])
    dirs = np.array([[0.5, 0.5, 0.5], [0.5, 0, 0.5], [0, 0.5, 0.5]])

    heights = np.array([5, 7, 10])

    actor_list = [[actor.cone, {'heights': heights, 'resolution': 8}],
                  [actor.arrow, {'heights': heights, 'resolution': 9}],
                  [actor.cylinder, {'heights': heights, 'resolution': 10}],
                  [actor.disk, {'rinner': 4, 'router': 8, 'rresolution': 2,
                                'cresolution': 2}]]

    scene = window.Scene()

    for act_func, extra_args in actor_list:
        colors = np.array([[1, 0, 0, 0.3], [0, 1, 0, 0.4], [1, 1, 0, 1]])

        geom_actor = act_func(xyz, dirs, colors, **extra_args)
        scene.add(geom_actor)

        if interactive:
            window.show(scene, order_transparent=True)
        arr = window.snapshot(scene)
        report = window.analyze_snapshot(arr, colors=colors)
        npt.assert_equal(report.objects, 3)

        scene.clear()

        colors = np.array([1.0, 1.0, 1.0, 1.0])

        geom_actor = act_func(xyz, dirs, colors, **extra_args)
        scene.add(geom_actor)

        if interactive:
            window.show(scene, order_transparent=True)
        arr = window.snapshot(scene)
        report = window.analyze_snapshot(arr, colors=[colors])
        npt.assert_equal(report.objects, 3)

        scene.clear()
```

Reply:
Agreed, looks nice.  I've updated the branch.